### PR TITLE
SQL export bug fix and always cleanup on script exit

### PR DESCRIPTION
**Pull request by wunderk1nd-e**
Fixes a bug on the sql export command that causes the script to exit early if the dump happens too quickly.
Also moves all of the cleanup code into an on exit trap.

### CODE REVIEW

File: cloud_sql_backup.sh [hunk 1]
```diff
@@ -14,6 +14,36 @@
...
+  echo '|'
+  echo '==================================================================================================='
+  echo
+
+  echo "Deleting ephemeral db instance used for backup: $TARGET_BACKUP_INSTANCE"
+  if [[ $TARGET_BACKUP_INSTANCE == *"backup"* ]]; then
+    gcloud -q sql instances delete "$TARGET_BACKUP_INSTANCE"
+  else
+    echo "String 'backup' not detected in target backup instance. Not deleting anything.."
+  fi
```
Comment (eversC) [inline cloud_sql_backup.sh]: some of these `echo`'s were previously using the `echo_out` (admittedly badly named) function, would we be able to keep using it? I've found it useful as it prefixes a `[<timestamp>]`

Comment (wunderk1nd-e) [reply cloud_sql_backup.sh]: Whoops - sorry about that, will sort.

Comment (wunderk1nd-e) [reply cloud_sql_backup.sh]: Sorted

**Comment by wunderk1nd-e** [comment]
> `cleanup()` should only be called after the permission grant and ephemeral db instance has been created. Here's what happens when I run the script with none of the required env vars set:
> 
> ```
> $ ./cloud_sql_backup.sh
> 
> DB_VERSION is required
> DB_NAME is required
> INSTANCE_CPU is required
> INSTANCE_ENV is required
> INSTANCE_MEM is required
> INSTANCE_NAME_PREFIX is required
> INSTANCE_REGION is required
> INSTANCE_STORAGE_SIZE_GB is required
> INSTANCE_STORAGE_TYPE is required
> PROJECT is required
> SA_KEY_FILEPATH is required
> SOURCE_BACKUP_INSTANCE is required
> TARGET_BACKUP_BUCKET is required
> 
> ===================================================================================================
> |
> | Revoking the new DB instances service account permission to write to GCS bucket
> |
> ===================================================================================================
> 
> [2019-10-08_10:18:39] Removing write access on  for 
> CommandException: "acl" command does not support "file://" URLs. Did you mean to use a gs:// URL?
> ```

Now registering the trap after the database has been created

**Pull request merged**